projects/RabbitMQ.Client/client/impl/SessionManager.cs

```diff
@@ -99,7 +99,7 @@ public ISession CreateInternal(int channelNumber)
         {
             lock (_sessionMap)
             {
-                ISession session = new Session(_connection, channelNumber);
+                ISession session = new Session(_connection, (ushort)channelNumber);
```

Comment:
Since we are now casting to ushort wouldn't it make sense to go one step further to change the `SessionManager`, `IntAllocator` and the rest of the infra as well to get rid of the cast?

Reply:
`IntAllocator` can be generalised to be a `ushort` allocator :)

Reply:
Yes that's possible. I didn't want to change too much in one go.
Also I think there is some public API that exposes it as int, so this is most likely going to stay.